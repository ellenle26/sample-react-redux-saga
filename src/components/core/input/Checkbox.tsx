import classNames from "classnames";
import { FC } from "react";

const Checkbox: FC<CheckboxProps> = (props: CheckboxProps) => {


    return (
        <div className={classNames("inline-block mt-3 mr-4", { "opacity-50": props.disabled }, props.className)}>
            <label className={"cursor-pointer leading-loose flex items-center"}>
                <input
                    id="checkbox"
                    className={"hidden"}
                    type="checkbox"
                    disabled={props.disabled ? props.disabled : false}
                    onChange={(event) => props.onChange && props.onChange(event.target.checked)}
                />
                <img
                    className={"inline-block mr-1"}
                    src={
                        props.checked
                            ? "/icons/checkbox_checked.svg"
                            : "/icons/checkbox_unchecked.svg"
                }
                    alt="icon"
                />
                {props.label && (
                    <p className={"text-sm text-gray-800"}>{props.label}</p>
                )}
            </label>
        </div>
    );
};

export interface CheckboxProps {
    checked?: boolean;
    label?: string;
    disabled?: boolean;
    className?: string;
    onChange?: (value: boolean) => void;
}

export default Checkbox;