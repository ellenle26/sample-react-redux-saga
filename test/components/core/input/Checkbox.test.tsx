/**
 * @jest-environment jsdom
*/

import Checkbox from "@components/core/input/Checkbox";
import { fireEvent, render } from "@testing-library/react";
import React from "react";

test("snapshot testing", () => {
    const renderResult = render(<Checkbox checked={false} />);
    expect(renderResult.asFragment()).toMatchSnapshot();
});

test("onchange", () => {
    const handleChange = jest.fn();
    const renderResult = render(<Checkbox checked={false} onChange={value => handleChange(value)} label="item"/>);

    fireEvent.click(renderResult.container.getElementsByTagName("input")[0]);
    expect(renderResult.container.getElementsByTagName("input")[0].checked).toBeTruthy();
    fireEvent.click(renderResult.container.getElementsByTagName("input")[0])
    expect(renderResult.container.getElementsByTagName("input")[0].checked).toBeFalsy()
})