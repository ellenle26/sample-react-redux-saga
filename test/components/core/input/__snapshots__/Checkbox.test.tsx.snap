// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="inline-block mt-3 mr-4"
  >
    <label
      class="cursor-pointer leading-loose flex items-center"
    >
      <input
        class="hidden"
        id="checkbox"
        type="checkbox"
      />
      <img
        alt="icon"
        class="inline-block mr-1"
        src="/icons/checkbox_unchecked.svg"
      />
    </label>
  </div>
</DocumentFragment>
`;
